src/Enzyme.jl

```diff
@@ -1006,64 +1006,75 @@ end
     return (one(x),)
 end
 
-"""
-    gradient(::ReverseMode, f, x)
+@inline function chunkedonehot(x, ::Val{chunk}) where chunk
+    sz = length(x)
+    num = ((sz + chunk - 1) ÷ chunk)
+    ntuple(Val(num)) do i
+        Base.@_inline_meta
+        onehot(x, (i-1)*chunk+1, i == num ? sz : (i*chunk) )
+    end
+end
 
-Compute the gradient of a real-valued function `f` using reverse mode.
-This will allocate and return new array `make_zero(x)` with the gradient result.
+@inline function chunkedonehot(x::AbstractFloat, ::Val{chunk}) where chunk
+    return ((one(x),),)
+end
 
-Besides arrays, for struct `x` it returns another instance of the same type,
-whose fields contain the components of the gradient.
-In the result, `grad.a` contains `∂f/∂x.a` for any differential `x.a`,
-while `grad.c == x.c` for other types.
+@inline tupleconcat(x) = x
+@inline tupleconcat(x, y) = (x..., y...)
+@inline tupleconcat(x, y, z...) = (x..., tupleconcat(y, z...)...)
 
-Examples:
+@inline function derivative(mode::ForwardMode, f, x; shadow=onehot(x))
```

Comment:
Since this is a new method, can we add tests?

Reply:
done